test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc

```diff
@@ -363,6 +474,128 @@ TEST_P(GrpcJsonTranscoderIntegrationTest, InvalidJson) {
       R"({ "theme"  "Children" })", {}, {}, Status(),
       Http::TestHeaderMapImpl{{":status", "400"}, {"content-type", "text/plain"}},
       "Expected : between key:value pair.\n", false);
+
+  testTranscoding<bookstore::CreateShelfRequest, bookstore::Shelf>(
+      Http::TestHeaderMapImpl{{":method", "POST"}, {":path", "/shelf"}, {":authority", "host"}},
+      R"({ "theme" : "Children" }EXTRA)", {}, {}, Status(),
+      Http::TestHeaderMapImpl{{":status", "400"}, {"content-type", "text/plain"}},
+      "Parsing terminated before end of input.\n", false);
+}
+
+std::string createDeepJson(int level, bool valid) {
+  std::string begin = R"({"k":)";
+  std::string deep_val = R"("v")";
+  std::string end = R"(})";
+  std::string json;
+
+  for (int i = 0; i < level; ++i) {
+    absl::StrAppend(&json, begin);
+  }
+  if (valid) {
+    absl::StrAppend(&json, deep_val);
+  }
+  for (int i = 0; i < level; ++i) {
+    absl::StrAppend(&json, end);
+  }
+  return json;
+}
+
+std::string jsonStrToPbStrucStr(std::string json) {
+  Envoy::ProtobufWkt::Struct message;
+  std::string structStr;
+  TestUtility::loadFromJson(json, message);
+  TextFormat::PrintToString(message, &structStr);
+  return structStr;
+}
+
+TEST_P(GrpcJsonTranscoderIntegrationTest, DeepStruct) {
+  HttpIntegrationTest::initialize();
+  // Due to the limit of protobuf util, we can only compare to level 32.
+  std::string deepJson = createDeepJson(32, true);
+  std::string deepProto = "content {" + jsonStrToPbStrucStr(deepJson) + "}";
```

Comment:
same here, less string constructing is preferred.

Reply:
I see. I can do that for createLargeJson but for createDeepJson, it also creates invalid Json, which in the very deep, has {k:}. I guess it may be generated from proto easily.